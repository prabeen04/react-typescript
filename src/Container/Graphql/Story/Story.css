.story-title{
    color: #888;
    font-size: 1rem;
    padding: 1rem;
    cursor: pointer;
    border: 1px solid #333;
}
.story-title:hover {
    color: rgb(179, 174, 174);
    border: 1px solid rgb(219, 214, 214);
}