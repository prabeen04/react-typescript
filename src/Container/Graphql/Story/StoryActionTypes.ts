export const TOGGLE_ADD_STORY_MODAL = 'TOGGLE_ADD_STORY_MODAL';
export const SET_SELECTED_STORY = 'SET_SELECTED_STORY';