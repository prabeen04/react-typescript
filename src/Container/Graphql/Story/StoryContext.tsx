import * as React from 'react';
const StoryContext = React.createContext({})
export default StoryContext