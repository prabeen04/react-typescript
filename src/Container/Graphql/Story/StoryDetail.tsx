import * as React from 'react'

export default function StoryDetail() {
  return (
    <>
      
    </>
  )
}
