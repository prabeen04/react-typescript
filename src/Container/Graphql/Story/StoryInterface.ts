
export interface IStories {
    addStoryModal: Boolean;
    stories: IStory[];
    storyDetail: IStory;
}

export interface IStory {
    title: String
}
