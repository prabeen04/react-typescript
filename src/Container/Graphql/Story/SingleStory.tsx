import * as React from 'react';
import { IStory } from './StoryInterface';

export default function Story(props: IStory): JSX.Element {
    return <h1 className='story-title' data-test="Story">{props.title}</h1>
}