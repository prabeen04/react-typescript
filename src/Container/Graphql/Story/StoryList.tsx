import * as React from 'react'
import { Query } from "react-apollo";
import gql from 'graphql-tag';
import Story from './Story';
import { IStory } from "./StoryInterface";
const SampleQuery = () => (
    <>
        <Query
            query={gql`
                {
                    stories {
                    title
                }
                }
            `}
        >
            {({ loading, error, data }) => {
                console.log(data)
                if (loading) return <p>Loading...</p>;
                if (error) return <p>Error :(</p>;

                return (
                    <div>
                        {
                            data.stories.map((item: IStory, i: any) => <Story key={i} title={item.title} />)
                        }
                    </div>
                );
            }}
        </Query>
    </>
);
export default SampleQuery;
