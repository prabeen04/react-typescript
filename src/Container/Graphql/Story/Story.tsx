import * as React from 'react'
import StoryList from "./StoryList";
import './Story.css';

export default function Story(): JSX.Element {
    return (
        <div>
            <StoryList />
        </div>
    )
}
