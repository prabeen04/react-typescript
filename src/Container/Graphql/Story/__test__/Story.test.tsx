import * as React from 'react';
import { shallow } from "enzyme";
import Story from '../Story';

describe('<Story/>', () => {
    it('should match snapshot', () => {
        const wrapper = shallow(<Story />)
        expect(wrapper).toMatchSnapshot()
    })
    it('should render story', () => {
        const wrapper = shallow(<Story />)
        expect(wrapper.find('[data-test="Story"]').length).toEqual(1)
    })
})
