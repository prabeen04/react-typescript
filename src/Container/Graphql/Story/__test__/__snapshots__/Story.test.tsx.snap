// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Story/> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Story />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <div
        className="story-wrapper"
        data-test="Story"
      >
        <div
          className="story-list"
        >
          <StoryList />
        </div>
        <div
          className="story-detail"
        >
          <StoryDetail />
        </div>
        <AddStoryModal />
      </div>,
      "value": Object {
        "dispatch": [Function],
        "state": Object {
          "addStoryModal": false,
        },
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="story-list"
          >
            <StoryList />
          </div>,
          <div
            className="story-detail"
          >
            <StoryDetail />
          </div>,
          <AddStoryModal />,
        ],
        "className": "story-wrapper",
        "data-test": "Story",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <StoryList />,
            "className": "story-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <StoryDetail />,
            "className": "story-detail",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": Object {
          "addStoryModal": false,
        },
        "_currentValue2": Object {
          "addStoryModal": false,
        },
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <div
          className="story-wrapper"
          data-test="Story"
        >
          <div
            className="story-list"
          >
            <StoryList />
          </div>
          <div
            className="story-detail"
          >
            <StoryDetail />
          </div>
          <AddStoryModal />
        </div>,
        "value": Object {
          "dispatch": [Function],
          "state": Object {
            "addStoryModal": false,
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="story-list"
            >
              <StoryList />
            </div>,
            <div
              className="story-detail"
            >
              <StoryDetail />
            </div>,
            <AddStoryModal />,
          ],
          "className": "story-wrapper",
          "data-test": "Story",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <StoryList />,
              "className": "story-list",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <StoryDetail />,
              "className": "story-detail",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": Object {
            "addStoryModal": false,
          },
          "_currentValue2": Object {
            "addStoryModal": false,
          },
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
